xml/cl.xml

```diff
@@ -382,6 +384,15 @@ server's OpenCL/api-docs repository.
             <member>const <type>size_t</type>*                             <name>global_work_size</name></member>
             <member>const <type>size_t</type>*                             <name>local_work_size</name></member>
         </type>
+
+        <type category="define" requires="stdint" name="CL_ICD2_TAG_KHR" comment="ICD 2 tag, in cl_ext.h">
+#if INTPTR_MAX == INT32_MAX
+#define CL_ICD2_TAG_KHR ((intptr_t)0x434C3331)
+#else
+#define CL_ICD2_TAG_KHR ((intptr_t)0x4F50454E434C3331)
+#endif
+        </type>
```

Comment:
This is different from how other `#define` constants are defined, making it very hard to read this definition automatically.

How about we instead specify what happens to upper bits in a comment?
```suggestion
        <type category="define" requires="stdint" comment="ICD 2 tag, in cl_ext.h; upper bits are cut off on 32-bit systems">#define <name>CL_ICD2_TAG_KHR</name> ((intptr_t)0x4F50454E434C3331)</type>
```

Reply:
That was my original idea. Unfortunately, since this is downcasting an integer it generates warning on some compilers. I checked, and Vulkan also use these king of logic:

```xml
        <type category="define" name="VK_USE_64_BIT_PTR_DEFINES">
#ifndef VK_USE_64_BIT_PTR_DEFINES
    #if defined(__LP64__) || defined(_WIN64) || (defined(__x86_64__) &amp;&amp; !defined(__ILP32__) ) || defined(_M_X64) || defined(__ia64) || defined (_M_IA64) || defined(__aarch64__) || defined(__powerpc64__) || (defined(__riscv) &amp;&amp; __riscv_xlen == 64)
        #define VK_USE_64_BIT_PTR_DEFINES 1
    #else
        #define VK_USE_64_BIT_PTR_DEFINES 0
    #endif
#endif</type>
```

Reply:
Hmm... What about `(intptr_t)(int64_t)0x4F50454E434C3331`? The idea is that it shouldn't generate a warning if the downcast is more clearly intended...

Reply:
That's what I though, but the warning was still there. And someone warned me it would happen beforehand, like you I was in disbelief.

Reply:
Alright, then, I agree with this solution; that's better than what I can suggest...